### PR TITLE
Refactored CognitoIdentity code examples

### PR DESCRIPTION
**Pull request by Doug-AWS**
*Issue #, if available:*
Cognito Identity Pool has one code example, list-identity-pools.

*Description of changes:*
Added:
- describe-identity-pool, which displays info about a single pool.
- list-pool-identities, which lists the identities in a pool.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        println!("Identitities:");
+        for id in ids {
+            let creation_date = id.creation_date.unwrap().to_system_time().unwrap();
+            let creation_datetime = DateTime::<Utc>::from(creation_date);
+            // Formats the combined date and time with the specified format string.
+            let creation_timestamp_str =
+                creation_datetime.format("%Y-%m-%d %H:%M:%S.%f").to_string();
+
+            let idid = id.identity_id.unwrap_or_default();
+            let mod_date = id.last_modified_date.unwrap().to_system_time().unwrap();
```
Comment (rcoh) [inline aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:86]: you can use `last_modified_date.to_chrono()`

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            let mod_datetime = DateTime::<Utc>::from(mod_date);
+            let mod_timestamp_str = mod_datetime.format("%Y-%m-%d %H:%M:%S.%f").to_string();
+
+            println!("  Creation data:      {}", creation_timestamp_str);
+            println!("  ID:                 {}", idid);
+            println!("  Last modified data: {}", mod_timestamp_str);
+
+            if let Some(logins) = id.logins {
```
Comment (rcoh) [inline aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:94]: probably just want `unwrap_or_default()` here

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            // Formats the combined date and time with the specified format string.
+            let creation_timestamp_str =
+                creation_datetime.format("%Y-%m-%d %H:%M:%S.%f").to_string();
+
+            let idid = id.identity_id.unwrap_or_default();
+            let mod_date = id.last_modified_date.unwrap().to_system_time().unwrap();
+            let mod_datetime = DateTime::<Utc>::from(mod_date);
+            let mod_timestamp_str = mod_datetime.format("%Y-%m-%d %H:%M:%S.%f").to_string();
+
+            println!("  Creation data:      {}", creation_timestamp_str);
```
Comment (rcoh) [inline aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:90]:
```suggestion
            println!("  Creation date:      {}", creation_timestamp_str);
```

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            println!("  ID:                 {}", idid);
+            println!("  Last modified data: {}", mod_timestamp_str);
```
Comment (rcoh) [inline aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:92]:
```suggestion
            println!("  Last modified date: {}", mod_timestamp_str);
```

**Review by rcoh — CHANGES_REQUESTED** [review]

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs (hunk 1 above)
Comment (Doug-AWS) [reply aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:86]: Awesome, thanks!

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs (hunk 3 above)
Comment (Doug-AWS) [reply aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:90]: Good catch.

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs (hunk 2 above)
Comment (Doug-AWS) [reply aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:94]: Ok

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs (hunk 4 above)
Comment (Doug-AWS) [reply aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:92]: Fixed

File: aws/sdk/examples/cognitoidentity/src/bin/list-identity-pools.rs [hunk 5]
```diff
@@ -4,27 +4,26 @@
...
     region: Option<String>,
 
-    /// Whether to display additional information
+    /// Whether to display additional information.
     #[structopt(short, long)]
     verbose: bool,
 }
 
-/// Lists your Amazon Cognito identities
+/// Lists your Amazon Cognito identitiy pools in the Region.
```
Comment (jdisanti) [inline aws/sdk/examples/cognitoidentity/src/bin/list-identity-pools.rs:21]:
```suggestion
/// Lists your Amazon Cognito identity pools in the Region.
```

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    /// The ID of the identity pool to describe.
+    #[structopt(short, long)]
+    identity_pool_id: String,
+
+    /// Whether to display additional information.
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+/// Lists the identities in an Amazon Cognito identitiy pool.
```
Comment (jdisanti) [inline aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:25]:
```suggestion
/// Lists the identities in an Amazon Cognito identity pool.
```

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                println!("    {}", login);
+            }
+
+            println!();
+        }
+    }
+
+    let next_token = response.next_token;
+
+    println!("Next token: {:?}", next_token);
```
Comment (jdisanti) [inline aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:90]:
```suggestion
    println!("Next token: {:?}", response.next_token);
```

**Review by jdisanti — APPROVED** [review]
LGTM. Minor fixes suggested.

**Comment by rcoh** [comment]
re-added label for John's changes

File: aws/sdk/examples/cognitoidentity/src/bin/list-identity-pools.rs (hunk 5 above)
Comment (Doug-AWS) [reply aws/sdk/examples/cognitoidentity/src/bin/list-identity-pools.rs:21]: Fixed

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs (hunk 6 above)
Comment (Doug-AWS) [reply aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:25]: Fixed

File: aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs (hunk 7 above)
Comment (Doug-AWS) [reply aws/sdk/examples/cognitoidentity/src/bin/list-pool-identities.rs:90]: Done.

**Review by jdisanti — APPROVED** [review]

**Pull request merged**